docs/topics/proc_exposing-application-metrics-using-prometheus-with-vertx.adoc

```diff
@@ -95,64 +95,86 @@ import io.micrometer.core.instrument.Meter;
 import io.micrometer.core.instrument.config.MeterFilter;
 import io.micrometer.core.instrument.distribution.DistributionStatisticConfig;
 import io.micrometer.prometheus.PrometheusMeterRegistry;
-import io.vertx.core.AbstractVerticle;
 import io.vertx.core.Vertx;
 import io.vertx.core.VertxOptions;
 import io.vertx.core.http.HttpServerOptions;
 import io.vertx.micrometer.MicrometerMetricsOptions;
 import io.vertx.micrometer.VertxPrometheusOptions;
 import io.vertx.micrometer.backends.BackendRegistries;
 
-public class MyVertxApp extends AbstractVerticle {
-
-    @Override
-    public void start() {
-        Vertx vertx = Vertx.vertx(new VertxOptions().setMetricsOptions(
-            new MicrometerMetricsOptions()
-                .setPrometheusOptions(new VertxPrometheusOptions().setEnabled(true)
-                    .setStartEmbeddedServer(true)
-                    .setEmbeddedServerOptions(new HttpServerOptions().setPort(8081))
-                    .setEmbeddedServerEndpoint("/metrics"))
-                .setEnabled(true)));
-
-        PrometheusMeterRegistry registry = (PrometheusMeterRegistry) BackendRegistries.getDefaultNow();
-        registry.config().meterFilter(
-            new MeterFilter() {
-                @Override
-                public DistributionStatisticConfig configure(Meter.Id id, DistributionStatisticConfig config) {
-                return DistributionStatisticConfig.builder()
-                    .percentilesHistogram(true)
-                    .build()
-                    .merge(config);
-                }
-            });
-
-        vertx.deployVerticle(MyVertxApp.class.getName());
-
-        Timer timer = Timer
-            .builder("my.timer")
-            .description("a description of what this timer does")
-            .register(registry);
-
-        vertx.setPeriodic(1000, l -> {
-            timer.record(() -> {
-                vertx.createHttpServer()
-                .requestHandler(req -> req.response().end("Hello"))
-                .listen(8080);
-            });
-        });
-    }
+public class CustomLauncher extends Launcher {
+
```

Comment:
IIRC, in custom `Launcher` class, you have to also provide `public static void main` with `dispatch()` method invocation in order to use it. Something like this:
```java
public static void main(String[] args) {
    new CustomLauncher().dispatch(args);
}
```

@tsegismont can you confirm? Thanks :)

Reply:
@spisiakm no it's not necessary.